includes/specials/SpecialUpload.php

```diff
@@ -110,7 +110,7 @@ protected function loadRequest() {
 
 		// If it was posted check for the token (no remote POST'ing with user credentials)
 		$token = $request->getVal( 'wpEditToken' );
-		$this->mTokenOk = $this->getUser()->matchEditToken( $token );
+		$this->mTokenOk = !is_null( $token ) && $this->getUser()->matchEditToken( $token ); # Wikia change
```

Comment:
Added `!is_null` checks before both calls to `hash_equals` in User class - 8b705d8